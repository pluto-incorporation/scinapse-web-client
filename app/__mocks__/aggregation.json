{
  "data": {
    "years": [
      { "year": 2008, "doc_count": 0 },
      { "year": 2009, "doc_count": 25 },
      { "year": 2010, "doc_count": 14 },
      { "year": 2011, "doc_count": 13 },
      { "year": 2012, "doc_count": 13 },
      { "year": 2013, "doc_count": 12 },
      { "year": 2014, "doc_count": 13 },
      { "year": 2015, "doc_count": 0 },
      { "year": 2016, "doc_count": 0 },
      { "year": 2017, "doc_count": 0 },
      { "year": 2018, "doc_count": 0 }
    ],
    "journals": [
      { "id": 25293849, "title": "The FASEB Journal", "doc_count": 2 },
      { "id": 107775873, "title": "Hydrobiologia", "doc_count": 2 },
      { "id": 3880285, "title": "Science", "doc_count": 1 },
      { "id": 11380747, "title": "Canadian Veterinary Journal-revue Veterinaire Canadienne", "doc_count": 1 },
      { "id": 11973042, "title": "Plant Archives", "doc_count": 1 },
      { "id": 12065667, "title": "Mikrochimica Acta", "doc_count": 1 },
      { "id": 13275601, "title": "American Scholar", "doc_count": 1 },
      { "id": 20845397, "title": "Journal of The Canadian Dental Association", "doc_count": 1 },
      { "id": 24807848, "title": "Physical Review Letters", "doc_count": 1 },
      { "id": 26277462, "title": "International Journal of Agriculture and Biology", "doc_count": 1 }
    ],
    "impact_factors": [
      { "from": 0, "to": 1, "doc_count": 9 },
      { "from": 1, "to": 2, "doc_count": 9 },
      { "from": 2, "to": 3, "doc_count": 7 },
      { "from": 3, "to": 4, "doc_count": 2 },
      { "from": 4, "to": 5, "doc_count": 2 },
      { "from": 5, "to": 6, "doc_count": 2 },
      { "from": 6, "to": 7, "doc_count": 0 },
      { "from": 7, "to": 8, "doc_count": 0 },
      { "from": 8, "to": 9, "doc_count": 1 },
      { "from": 9, "to": 10, "doc_count": 0 },
      { "from": 10, "to": null, "doc_count": 32 }
    ],
    "fos_list": [
      { "id": 127413603, "name": "Engineering", "doc_count": 33 },
      { "id": 71924100, "name": "Medicine", "doc_count": 29 },
      { "id": 185592680, "name": "Chemistry", "doc_count": 24 },
      { "id": 41008148, "name": "Computer Science", "doc_count": 22 },
      { "id": 121332964, "name": "Physics", "doc_count": 21 },
      { "id": 86803240, "name": "Biology", "doc_count": 20 },
      { "id": 192562407, "name": "Materials Science", "doc_count": 14 },
      { "id": 17744445, "name": "Political Science", "doc_count": 13 },
      { "id": 33923547, "name": "Mathematics", "doc_count": 12 },
      { "id": 46141821, "name": "Nuclear magnetic resonance", "doc_count": 9 }
    ]
  }
}
