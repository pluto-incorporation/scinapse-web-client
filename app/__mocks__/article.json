{
  "id": 20,
  "type": "POST_PAPER",
  "title": "this is title",
  "articleAbstract": "this is abstract.",
  "summary": "this is summary.",
  "link": "this is summary.",
  "source": "ARXIV",
  "articlePublishedAt": "2017-10-10T10:10:10",
  "articleUpdatedAt": "2017-10-10T10:10:10",
  "point": {
    "total": 7,
    "originality": 7,
    "contribution": 7,
    "analysis": 7,
    "expressiveness": 7
  },
  "createdBy": {
    "id": 10,
    "email": "test",
    "name": "tedyeee",
    "reputation": 16
  },
  "createdAt": "2017-10-09T10:47:32.105",
  "authors": [
    {
      "id": 30,
      "type": "LEAD_AUTHOR",
      "name": "pepsi",
      "organization": "Coca Cola"
    },
    {
      "id": 31,
      "member": {
        "id": 10,
        "email": "test",
        "name": "tedyeee",
        "reputation": 16
      },
      "type": "CORRESPONDING_AUTHOR",
      "name": "coke",
      "organization": "kingsi"
    }
  ],
  "evaluations": [
    {
      "id": 50,
      "point": {
        "total": 7,
        "originality": 7,
        "contribution": 7,
        "analysis": 7,
        "expressiveness": 7,
        "originalityComment": "originality comment",
        "contributionComment": "contribution comment",
        "analysisComment": "analysis comment",
        "expressivenessComment": "expressiveness comment"
      },
      "vote": 1,
      "comments": [
        {
          "id": 60,
          "createdBy": {
            "id": 10,
            "email": "test",
            "name": "tedyeee",
            "reputation": 16
          },
          "comment": "this is comment.",
          "createdAt": "2017-10-09T10:49:26.719"
        }
      ],
      "createdBy": {
        "id": 10,
        "email": "test",
        "name": "tedyeee",
        "reputation": 16
      },
      "createdAt": "2017-10-09T10:48:22.298"
    }
  ]
}
