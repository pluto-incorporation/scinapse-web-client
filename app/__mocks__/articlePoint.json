{ "total": 6.75, "originality": 4.0, "significance": 6.0, "validity": 8.0, "organization": 9.0 }
