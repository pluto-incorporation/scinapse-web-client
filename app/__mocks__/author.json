{
  "id": 2258681040,
  "name": "H. Goldschmidt",
  "email": "scshinjr@gmail.com",
  "bio":
    "Local server runs with Express and webpack-dev-server.\nBecause Scinapse is the universal rendering web-app, it needs both server-side app file and client-side app file. \nLocal server runs with Express and webpack-dev-server.\nBecause Scinapse is the universal rendering web-app, it needs both server-side app file and client-side app file.\nLocal server runs with Express and webpack-dev-server.\nBecause Scinapse is the universal rendering web-app, it needs both server-side app file and client-side app file.\nLocal server runs with Express and webpack-dev-server.\nBecause Scinapse is the universal rendering web-app, it needs both server-side app file and client-side app file.",
  "webPage": null,
  "hindex": null,
  "last_known_affiliation": { "id": 223822909, "name": "Heidelberg University" },
  "paper_count": 19,
  "citation_count": 3398,
  "is_layered": true,
  "representative_papers": [],
  "fos_list": [],
  "top_papers": [],
  "profile_id": null,
  "is_profile_connected": false
}
