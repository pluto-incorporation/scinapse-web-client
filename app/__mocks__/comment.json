{
  "id": 60,
  "comment": "this is comment.",
  "createdBy": {
    "id": 10,
    "email": "test@pluto.network",
    "name": "tedyeee",
    "profileImage": null,
    "reputation": 21,
    "wallet": null
  },
  "createdAt": "2017-10-11T04:34:54.705"
}
