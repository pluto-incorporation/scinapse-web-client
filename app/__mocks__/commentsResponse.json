{
  "content": [
    {
      "id": 108,
      "comment": "sample comment.",
      "paperId": null,
      "cognitivePaperId": 2108376819,
      "createdBy": {
        "id": 52,
        "email": "taeheon@pluto.network",
        "emailVerified": true,
        "name": "asdf",
        "profileImage": null,
        "affiliation": "asdf",
        "major": null,
        "reputation": 0,
        "articleCount": 0,
        "reviewCount": 0,
        "commentCount": 0,
        "oauth": null
      },
      "createdAt": "2018-03-06T02:14:08.356"
    },
    {
      "id": 107,
      "comment": "sample comment.",
      "paperId": null,
      "cognitivePaperId": 2108376819,
      "createdBy": {
        "id": 52,
        "email": "taeheon@pluto.network",
        "emailVerified": true,
        "name": "asdf",
        "profileImage": null,
        "affiliation": "asdf",
        "major": null,
        "reputation": 0,
        "articleCount": 0,
        "reviewCount": 0,
        "commentCount": 0,
        "oauth": null
      },
      "createdAt": "2018-03-06T02:14:07.838"
    },
    {
      "id": 106,
      "comment": "sample comment.",
      "paperId": null,
      "cognitivePaperId": 2108376819,
      "createdBy": {
        "id": 52,
        "email": "taeheon@pluto.network",
        "emailVerified": true,
        "name": "asdf",
        "profileImage": null,
        "affiliation": "asdf",
        "major": null,
        "reputation": 0,
        "articleCount": 0,
        "reviewCount": 0,
        "commentCount": 0,
        "oauth": null
      },
      "createdAt": "2018-03-06T02:14:07.356"
    },
    {
      "id": 105,
      "comment": "sample comment.",
      "paperId": null,
      "cognitivePaperId": 2108376819,
      "createdBy": {
        "id": 52,
        "email": "taeheon@pluto.network",
        "emailVerified": true,
        "name": "asdf",
        "profileImage": null,
        "affiliation": "asdf",
        "major": null,
        "reputation": 0,
        "articleCount": 0,
        "reviewCount": 0,
        "commentCount": 0,
        "oauth": null
      },
      "createdAt": "2018-03-06T02:14:06.814"
    },
    {
      "id": 104,
      "comment": "sample comment.",
      "paperId": null,
      "cognitivePaperId": 2108376819,
      "createdBy": {
        "id": 52,
        "email": "taeheon@pluto.network",
        "emailVerified": true,
        "name": "asdf",
        "profileImage": null,
        "affiliation": "asdf",
        "major": null,
        "reputation": 0,
        "articleCount": 0,
        "reviewCount": 0,
        "commentCount": 0,
        "oauth": null
      },
      "createdAt": "2018-03-06T02:14:06.274"
    },
    {
      "id": 103,
      "comment": "sample comment.",
      "paperId": null,
      "cognitivePaperId": 2108376819,
      "createdBy": {
        "id": 52,
        "email": "taeheon@pluto.network",
        "emailVerified": true,
        "name": "asdf",
        "profileImage": null,
        "affiliation": "asdf",
        "major": null,
        "reputation": 0,
        "articleCount": 0,
        "reviewCount": 0,
        "commentCount": 0,
        "oauth": null
      },
      "createdAt": "2018-03-06T02:14:05.758"
    }
  ],
  "totalElements": 22,
  "last": false,
  "totalPages": 4,
  "sort": null,
  "first": true,
  "numberOfElements": 6,
  "size": 6,
  "number": 0
}
