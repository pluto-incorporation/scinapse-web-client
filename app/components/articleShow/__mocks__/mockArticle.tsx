export function getMockArticle() {
  return {
    articleId: 1,
    tags: ["Open Access Paper", "CAU Paper"],
  };
}
