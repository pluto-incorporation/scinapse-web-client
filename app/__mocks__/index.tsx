import { IArticle, recordifyArticle } from "../model/article";
import { IWallet, WalletFactory } from "../model/wallet";
import { IAuthor, recordifyAuthor } from "../model/author";
import { IMember, recordifyMember } from "../model/member";
import { IArticlePoint, ArticlePointFactory } from "../model/articlePoint";
import { IComment, recordifyComment } from "../model/comment";
import { IReviewPoint, ReviewPointFactory } from "../model/reviewPoint";
import { IReview, recordifyReview } from "../model/review";

export const RAW = {
  ARTICLE: require("./article.json") as IArticle,
  ARTICLE_POINT: require("./articlePoint.json") as IArticlePoint,
  AUTHOR: require("./author.json") as IAuthor,
  COMMENT: require("./comment.json") as IComment,
  REVIEW: require("./review.json") as IReview,
  REVIEW_POINT: require("./reviewPoint.json") as IReviewPoint,
  MEMBER: require("./member.json") as IMember,
  WALLET: require("./wallet.json") as IWallet,
};

export const RECORD = {
  ARTICLE: recordifyArticle(RAW.ARTICLE),
  ARTICLE_POINT: ArticlePointFactory(RAW.ARTICLE_POINT),
  AUTHOR: recordifyAuthor(RAW.AUTHOR),
  COMMENT: recordifyComment(RAW.COMMENT),
  REVIEW: recordifyReview(RAW.REVIEW),
  REVIEW_POINT: ReviewPointFactory(RAW.REVIEW_POINT),
  MEMBER: recordifyMember(RAW.MEMBER),
  WALLET: WalletFactory(RAW.WALLET),
};
