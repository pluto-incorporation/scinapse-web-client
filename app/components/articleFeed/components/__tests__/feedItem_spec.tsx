jest.unmock("../article");

import * as React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import FeedItem, { IFeedItemProps } from "../feedItem";
import { RECORD } from "../../../../__mocks__/index";

describe("<FeedItem /> component", () => {
  let feedItemWrapper: ShallowWrapper<IFeedItemProps>;

  beforeEach(() => {
    feedItemWrapper = shallow(<FeedItem article={RECORD.ARTICLE} />);
  });

  it("should match snapshot", () => {
    expect(feedItemWrapper.html()).toMatchSnapshot();
  });
});
