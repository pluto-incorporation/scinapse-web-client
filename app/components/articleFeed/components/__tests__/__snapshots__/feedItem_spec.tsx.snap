// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedItem /> component should match snapshot 1`] = `"<div class=\\"feedItemWrapper\\"><div class=\\"contentSection\\"><div class=\\"leftBox\\"><div class=\\"tagWrapper\\"><div class=\\"tagListWrapper\\"><a class=\\"tagItem\\">POST PAPER</a></div></div><div class=\\"contentWrapper\\"><div class=\\"title\\">this is title.</div><div class=\\"abstractSummary\\">this is summary.</div></div></div><div class=\\"rightBox\\"><div class=\\"rightBoxContent\\"><div class=\\"evaluationPoint\\">7.4</div><div class=\\"evaluationPointText\\">pointed</div></div></div></div><div class=\\"informationBox\\"><div class=\\"informationLeftBox\\"><div><div class=\\"profileImageWrapper\\"><img class=\\"icon avatarIcon\\" src=\\"https://dd2gn9pwu61vr.cloudfront.net/avatar\\" alt=\\"AVATAR\\"/></div><div class=\\"authorInformation\\"><div class=\\"authorName\\">tedyeee</div><div class=\\"authorOrganization\\">test@pluto.network</div></div></div></div><div class=\\"informationRightBox\\"><div class=\\"createdAt\\">Oct 11, 2017</div></div></div></div>"`;
